src/status_im2/contexts/syncing/scan_sync_code/view.cljs

```diff
@@ -60,26 +62,37 @@
                         (reset! read-qr-once? false))}]]])
 
 (defn- camera-permission-view
-  [request-camera-permission]
-  [rn/view {:style style/camera-permission-container}
-   [quo/text
-    {:size   :paragraph-1
-     :weight :medium
-     :style  style/enable-camera-access-header}
-    (i18n/label :t/enable-access-to-camera)]
-   [quo/text
-    {:size   :paragraph-2
-     :weight :regular
-     :style  style/enable-camera-access-sub-text}
-    (i18n/label :t/to-scan-a-qr-enable-your-camera)]
-   [quo/button
-    {:before              :i/camera
-     :type                :primary
-     :size                32
-     :accessibility-label :request-camera-permission
-     :override-theme      :dark
-     :on-press            request-camera-permission}
-    (i18n/label :t/enable-camera)]])
+  [request-camera-permission perform-preflight-check]
+  (let [title-label-key       (if @camera-permission-granted?
```

Comment:
maybe slightly easier to read if you abstract the variables to a helper function..
```
(defn get-description [camera-permission-granted? request-camera-permission perform-preflight-check]
  (if camera-permission-granted?
    {:title-label-key       :t/enable-access-to-local-network
     :description-label-key :t/to-pair-with-your-other-device-in-the-network
     :button-icon          :i/world
     :button-label         :t/enable-network-access
     :on-press              perform-preflight-check}
    {:title-label-key       :t/enable-access-to-camera
     :description-label-key :t/to-scan-a-qr-enable-your-camera
     :button-icon           :i/camera
     :button-label         :t/enable-camera
     :on-press request-camera-permission}))

(let [{:keys [title-label-key
              description-label-key
              button-icon
              button-label
              on-press]} (get-description @camera-permission-granted?
                                          request-camera-permission
                                          perform-preflight-check)]
  ...)
  ```

Reply:
Good one 👍